kernel/ksud.c

```diff
@@ -163,24 +205,37 @@ int ksu_handle_execveat_ksud(int *fd, struct filename **filename_ptr,
 	}
 
 	if (unlikely(!memcmp(filename->name, system_bin_init,
-		    sizeof(system_bin_init) - 1))) {
```

Comment:
Don't bother the logic of Android 10+. We don't want to introduce new bugs for just adapting Android 10-

Reply:
You should make it a single condition block, don't modify code of Android 10+. If Android 6-7 needs to modify it, please drop.